aws/resource_aws_location_tracker.go

```diff
@@ -0,0 +1,159 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/locationservice"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/terraform-providers/terraform-provider-aws/aws/internal/keyvaluetags"
+)
+
+func resourceAwsLocationTracker() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsLocationTrackerCreate,
+		Read:   resourceAwsLocationTrackerRead,
+		Delete: resourceAwsLocationTrackerDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"kms_key_id": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Computed:     true,
+				ForceNew:     true,
+				ValidateFunc: validateArn,
+			},
+			"pricing_plan": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringInSlice(locationservice.PricingPlan_Values(), false),
+			},
+			"pricing_plan_data_source": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"Esri",
+					"Here",
+				}, false),
+			},
+			"tags": tagsSchemaForceNew(),
```

Comment:
not sure this should be force new, see [docs](https://github.com/hashicorp/terraform-provider-aws/blob/main/docs/contributing/contribution-checklists.md#resource-tagging-code-implementation) for updating tags in update func.
More specifically follow the below example:
```
if d.HasChange("tags_all") {
  o, n := d.GetChange("tags_all")
  if err := keyvaluetags.EksUpdateTags(conn, d.Get("arn").(string), o, n); err != nil {
    return fmt.Errorf("error updating tags: %s", err)
  }
}
```

Reply:
Currently Amazon Location Service does not provider API for modify tracker attributes:
https://docs.aws.amazon.com/location-trackers/latest/APIReference/API_CreateTracker.html

We'll need to add this functionality when the API becomes available.

Reply:
resources don't usually support updating tags in their Update action. Location does have a tag resource api available: 
https://docs.aws.amazon.com/cli/latest/reference/location/tag-resource.html